app/main.py

```diff
@@ -103,74 +112,116 @@ async def hello():
-            # launch each panels computation
-            soon_panels.append(task_group.soonify(obj_kp.hunger_game_kp)())
-            soon_panels.append(task_group.soonify(obj_kp.data_quality_kp)())
-            soon_panels.append(task_group.soonify(obj_kp.last_edits_kp)())
-            soon_panels.append(task_group.soonify(obj_kp.wikidata_kp)())
-        # collect panels results
-        panels = {}
-        for soon_value in soon_panels:
-            # Appending soon_value value in panels
-            # as soon_panels needs to access outside taskgroup.
-            if soon_value.value:
-                panels.update(soon_value.value)
-        return {"knowledge_panels": panels}
-
-
-templates = Jinja2Templates(directory="template")
+        with active_translation(lang_code.value):
+            # creating object that will compute knowledge panels
+
+            obj_kp = KnowledgePanels(
+                facet=facet_tag,
+                value=value_tag,
+                sec_facet=sec_facet_tag,
+                sec_value=sec_value_tag,
+                country=country.value if country is not Country.world else None,
+            )
+            # this will contains panels computations
+            soon_panels = []
+            # the task_group will run these knowledge_panels async functions concurrently
+            async with asyncer.create_task_group() as task_group:
+                # launch each panels computation
+                soon_panels.append(task_group.soonify(obj_kp.hunger_game_kp)())
+                soon_panels.append(task_group.soonify(obj_kp.data_quality_kp)())
+                soon_panels.append(task_group.soonify(obj_kp.last_edits_kp)())
+                soon_panels.append(task_group.soonify(obj_kp.wikidata_kp)())
+            # collect panels results
+            for soon_value in soon_panels:
+                # Appending soon_value value in panels
+                # as soon_panels needs to access outside taskgroup.
+                if soon_value.value:
+                    panels.update(soon_value.value)
+
+    if add_information_panels and value_tag is not None:
+        # As we're using user-provided data to access filesystem,
+        # generate secure filename
+        facet_tag_safe = secure_filename(facet_tag)
+        value_tag_safe = secure_filename(value_tag)
+
+        if facet_tag_safe and value_tag_safe:
+            file_path = find_kp_html_path(
+                HTML_DIR, facet_tag_safe, value_tag_safe, country, lang_code
+            )
+            panel = None
+            if file_path is not None:
+                async with async_open(file_path, "r") as f:
+                    html_content = await f.read()
+                panel = {
+                    "elements": [{"element_type": "text", "text_element": {"html": html_content}}],
+                    "title_element": {"title": "Description"},
+                }
+                panels["Description"] = panel
```

Comment:
why didn't you follow the pattern above and just add a function to do things asynchronously ? It seem to me it hinders readability to break the pattern.

Normally the add_information_panels can be check to decide to add or not the task to soon_panels.
And all the rest should be done inside the method (you can return None to have no panel added).